src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java

```diff
@@ -0,0 +1,435 @@
+		for (final AuthorizationIntraCloudResponseDTO rule : rules) {
+			for (final Long id : systemIds) {
+				if (rule.getConsumerSystem().getId() == id) {
+					result.add((rule.getId()));
+				}
+			}
+		}
+		return result;
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<SystemResponseDTO> getSystems() {
+		logger.debug("Get systems request started...");
+		final SystemListResponseDTO response = arrowheadService.consumeServiceHTTP(SystemListResponseDTO.class,
+				HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						serviceRegistryAddress,
+						serviceRegistryPort,
+						CommonConstants.SERVICEREGISTRY_URI + ConsumerConstants.QUERY_GET_SYSTEMS),
+				null, null);
+		return response.getData();
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<ServiceDefinitionResponseDTO> getServices() {
+		logger.debug("Get services request started...");
+		final ServiceDefinitionsListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				ServiceDefinitionsListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						serviceRegistryAddress,
+						serviceRegistryPort,
+						CommonConstants.SERVICEREGISTRY_URI + ConsumerConstants.QUERY_GET_SERVICES),
+				null, null);
+		return response.getData();
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<ServiceInterfaceResponseDTO> getInterfaces() {
+		logger.debug("Get interfaces request started...");
+		final ServiceInterfacesListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				ServiceInterfacesListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						serviceRegistryAddress,
+						serviceRegistryPort,
+						CommonConstants.SERVICEREGISTRY_URI + ConsumerConstants.QUERY_GET_INTERFACES),
+				null, null);
+		return response.getData();
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<AuthorizationIntraCloudResponseDTO> getAuthorizationRules() {
+		Map<String, String> authorizationUri = null;
+		List<AuthorizationIntraCloudResponseDTO> rules = null;
+		try {
+			authorizationUri = getAuthorizationUri();
+		} catch (final Exception e) {
+			logger.error(e.getMessage());
+		}
+		logger.debug("Get authorization rules request started...");
+		final AuthorizationIntraCloudListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				AuthorizationIntraCloudListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(authorizationUri.get(ConsumerConstants.SCHEME), authorizationUri.get(ConsumerConstants.HOST),
+						Integer.parseInt(authorizationUri.get(ConsumerConstants.PORT)),
+						authorizationUri.get(ConsumerConstants.PATH) + ConsumerConstants.QUERY_AUTH_INTRA_CLOUD),
+				null, null);
+
+		if (response == null) {
```

Comment:
Probably `response` never will be null (`arrowheadService.consumeServiceHTTP()` throws an excpetion if no response), but if I'm not right,  logging is not enough either. You should terminate if existing rules cannot be fetched.